app/src/engine/process/process_posix.hpp

```diff
@@ -121,18 +143,33 @@ class Process : public IProcess {
             // If it fails, it returns -1 and we throw an exception.
             // Is the _exit(0) necessary?
             // execv is replacing the current process with the new process
-            if (execv(command.c_str(), execv_argv) == -1) {
-                throw std::runtime_error("Failed to execute engine");
-            }
+            execv(command.c_str(), execv_argv);
 
-            _exit(0);
+            _exit(EXIT_FAILURE);
         }
         // This is the parent process
         else {
+            // wait until the file exists
+            const std::string base_path   = "/proc/";
+            const std::string status_file = "/status";
+            const auto path               = base_path + std::to_string(fork_pid) + status_file;
+
+            while (!fileReady(path)) {
+                std::this_thread::sleep_for(std::chrono::milliseconds(10));
+            }
+
+            // wait until the process is spawned and the name has been changed away from
+            // Name:   fast-chess to the actual name of the process
+            if (isZombie(path, "fast-chess")) {
```

Comment:
This will mean users can't change the name of the binary (like fast-chess.master, fast-chess.patch, which is common). You might be able to get it from argv 0 ?

Reply:
good point, will change that later and use something like this to detect if the process is working, since the proc/pid/status approach might not work for other platforms

```cpp
#include <stdio.h>
#include <sys/sysctl.h>
#include <sys/types.h>
#include <sys/user.h>
#include <unistd.h>
#include <stdlib.h>

int main() {
    int mib[4];
    size_t len;
    struct kinfo_proc *procs;
    int proc_count;

    mib[0] = CTL_KERN;
    mib[1] = KERN_PROC;
    mib[2] = KERN_PROC_ALL;
    mib[3] = 0;

    // First call to sysctl to get the size of the data to be returned
    sysctl(mib, 4, NULL, &len, NULL, 0);
    procs = (struct kinfo_proc *)malloc(len);

    // Second call to sysctl to get the actual process data
    sysctl(mib, 4, procs, &len, NULL, 0);

    proc_count = len / sizeof(struct kinfo_proc);
    for (int i = 0; i < proc_count; i++) {
        struct kinfo_proc *proc = &procs[i];

#ifdef __FreeBSD__
        if (proc->ki_stat == SZOMB) {
            printf("Process %d is a zombie\n", proc->ki_pid);
        }
#else
        if (proc->kp_proc.p_stat == SZOMB) {
            printf("Process %d is a zombie\n", proc->kp_proc.p_pid);
        }
#endif
    }

    free(procs);
    return 0;
}
```

Reply:
yeah, I think this is rather system dependent and thus fragile code.